### ISSUE TITLE
cog-delete-recursive documentation isn't correct

### ISSUE THREAD

**Issue by ngeiswei** [opencog/atomspace#127]
cog-delete-recursive's documentation isn't exactly correct here

``` scheme
       ; Verify that the link l is gone:
       guile> l
       Invalid handle

       ; Verify that the node x is gone:
       guile> x
       Invalid handle
```

in reality it shows the atoms referenced by l and x

``` scheme
scheme@(guile-user)> l
$8 = (Link
   (ConceptNode "abc")
   (ConceptNode "def")
)

scheme@(guile-user)> x
$9 = (ConceptNode "abc")
```

However the behavior of cog-delete-recursive is correct, using prt-atomspace as given in http://wiki.opencog.org/w/Scheme#Print_all_atoms_in_the_AtomSpace (EDIT: you can use directly cog-prt-atomspace) I can see that the atoms corresponding to l and x are indeed missing.

I'm not sure whether the documentation should be fixed or what is returned by l and x should be fixed.

**Comment by ngeiswei** [comment]
For those who don't know there exists an already coded cog-prt-atomspace, rather than using prt-atomspace for the wiki.

**Comment by linas** [comment]
Yeah, that's kind of weird.  Here is what is happening:

In the good old days, printing checked the UUID, and if it was -1, it printed "invalid handle".   Now, guile holds on to the atom pointer, and so the atom gets printed even if it is not in any atomspace. 

The correct fix is probably to alter the printing to also print (in a comment?) the atomspace that the atom belongs to.

We should add a new command `(cog-atomspace-of  SOME-ATOM)` that returns the atomspace that SOME-ATOM belongs to.

We probably should change `cog-delete` to `cog-remove`  and likewise for recursive, since it doesn't "delete", is simply removes the atom from the atomspace.

**Comment by linas** [comment]
Closing. Fixed in pull req #1341 Normally its considered polite to let the person who reported the bug to verify and close, but I figure Nil has his hands full, so I'm closing on his behalf.
